tests/host/Makefile

```diff
@@ -31,34 +50,64 @@ CORE_C_FILES := $(addprefix $(CORE_PATH)/,\
 	spiffs/spiffs_gc.c \
 	spiffs/spiffs_hydrogen.c \
 	spiffs/spiffs_nucleus.c \
+	libb64/cencode.c \
 )
 
-MOCK_CPP_FILES := $(addprefix common/,\
+MOCK_CPP_FILES_COMMON := $(addprefix common/,\
 	Arduino.cpp \
 	spiffs_mock.cpp \
 	WMath.cpp \
+	HardwareSerial.cpp \
+	MockTools.cpp \
+)
+
+MOCK_CPP_FILES := $(MOCK_CPP_FILES_COMMON) $(addprefix common/,\
+	ArduinoCatch.cpp \
+)
+
+MOCK_CPP_FILES_EMU := $(MOCK_CPP_FILES_COMMON) $(addprefix common/,\
+	ArduinoMain.cpp \
+	user_interface.cpp \
 )
 
 MOCK_C_FILES := $(addprefix common/,\
 	md5.c \
 	noniso.c \
 )
 
-INC_PATHS += $(addprefix -I, \
+INC_PATHS := $(addprefix -I,\
 	common \
 	$(CORE_PATH) \
 )
 
+INC_PATHS += $(addprefix -I,\
+	$(shell echo ../../libraries/*/src) \
+	$(shell echo ../../libraries/*) \
+	../../tools/sdk/include \
+	../../tools/sdk/lwip2/include \
+)
+
 TEST_CPP_FILES := \
 	fs/test_fs.cpp \
 	core/test_pgmspace.cpp \
 	core/test_md5builder.cpp \
 	core/test_string.cpp
 
-CXXFLAGS += -std=c++11 -Wall -Werror -coverage -O0 -fno-common -g
-CFLAGS += -std=c99 -Wall -Werror -coverage -O0 -fno-common -g
-LDFLAGS += -coverage -O0
+PREINCLUDES := \
+	-include common/mock.h \
+	-include common/c_types.h \
+
+ifneq ($(D),)
+DEBUG += -DDEBUG_ESP_PORT=Serial
+DEBUG += -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_MDNS
+endif
+
+CXXFLAGS += $(DEBUG) -std=c++11 -Wall -Werror -coverage -O0 -fno-common -g $(M32)
```

Comment:
I know the original makefile has `-O0` (oh-zero), but is there some reason for this?  Wouldn't `-Os` be closer to the real system?  Same w/the LDFLAGS.

Reply:
I'll add options to change that.
> run more of core through host-based testing and code analysis

I guess `-O0` was here for debugging with gdb on host and valgrind detailed report.